### PR TITLE
Add Retraining Helpers 

### PR DESCRIPTION
**Pull request by agude**
Added code to help retrain Resnet50.

### CODE REVIEW

**Comment by agude** [comment]
I think this bit is ready for merging. @tiffapedia or @ymt123 could you review?

File: pelops/training/utils.py [hunk 1]
```diff
@@ -105,3 +109,94 @@ def make_model_color(chip):
...
+        """ Writes links to a directory.
+
+        The final directory will be:
+
+        output_directory / root / class_number / image
+
+        Where root is set by self.__set_root_dir() and is based on
+        the SetType, but you can reset it by passing in root.
+
+        Args:
```
Comment (agude) [inline pelops/training/utils.py]: I'm a bad boy! :-1: I'll fix up this tomorrow.

**Comment by agude** [comment]
Alright, the missing doc string is now fixed; ready for review. @tiffapedia

File: pelops/training/utils.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+
+
+def attributes_to_classes(chip_dataset, value_extractor):
+    """Extract a set of attributes from a set of Chips and uses them to make
+    unique classses.
+
+    Args:
+        chip_dataset: A ChipDataset, or other iterable of Chips
+        value_extractor: A callable that takes a chip and returns a tuple
```
Comment (tiffapedia) [inline pelops/training/utils.py]: This doesn't strike out to me as very clear. I understand that value_extractor is a function that references, for example, the function make_model in utils.py. In the future, we might want to revisit this to make it more intuitive.

Comment (agude) [inline pelops/training/utils.py]: I'll try to make it clearer.

File: pelops/training/utils.py [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+        if root is None:
+            root = self.root
+
+        # Link chips
+        for chip in self.__chip_dataset:
+            src = chip.filepath
+            filename = os.path.basename(src)
+            chip_class = self.__chip_tuplizer(chip)
+            chip_index = self.__class_to_index[chip_class]
+            dest_dir = output_directory + "/" + root + "/" + str(chip_index) + "/"
```
Comment (ymt123) [inline pelops/training/utils.py]: Use os.path.join

File: pelops/training/utils.py [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+            os.makedirs(dest_dir, exist_ok=True)
+            os.link(src=src, dst=dest_dir + filename)
```
Comment (ymt123) [inline pelops/training/utils.py]: Use os.path.join (and maybe move to a separate line)

File: pelops/training/utils.py (hunk 3 above)
Comment (agude) [inline pelops/training/utils.py]: Wow, of course that function exists. I've added it in several places.

**Pull request merged**